app/harvesters/uva_harvester.rb

```diff
@@ -0,0 +1,248 @@
+    else
+      collections.map { |c| uri.gsub('[PID]', c) }
+    end
+  end
+
+  ##
+  # @see Krikri::Harvester.expected_opts
+  def self.expected_opts
+    {
+      key: :uva,
+      opts: {
+        collections: { type: :string, multiple_ok: true, required: false },
+        threads:  { type: :integer, required: false },
+        max_records:  { type: :integer, required: false }
+      }
+    }
+  end
+
+  ##
+  # @see Krikri::Harvester#count
+  def count
+    result = 0
+    @collection_harvesters.each { |harv| result += harv.count }
+    result
+  end
+
+  ##
+  # @return [Enumerator] an enumerator of the records targeted by this
+  #   harvester.
+  def records
+    Enumerator.new do |en|
+      all_collections = @collection_harvesters.map { |harv| harv.records }
+
+      all_collections.each do |collection|
+        collection.each do |record|
+          en << record
+        end
+      end
+    end
+  end
+
+  ##
+  # @param identifier [#to_s] the identifier of the record to get
+  # @return [#to_s] the record
+  def get_record(identifier)
+    @collection_harvesters.each do |harv|
+      rec = harv.get_record(identifier)
+      break rec if rec
+    end
+  end
+
+  ##
+  # Harvester for a single collection
+  class UVACollectionHarvester
+    include Krikri::Harvester
+
+    def initialize(uri, threads, max_records, record_class, id_minter)
+      @uri = uri
+      @threads = threads
+      @max_records = max_records
+      @record_class = record_class
+      @id_minter = id_minter
+      @http = Krikri::AsyncUriGetter.new
+    end
+
+    ##
+    # @see Krikri::Harvester#count
+    def count
+      Integer(records_mets.count)
+    end
+
+    ##
+    # @return [Enumerator::Lazy] an enumerator of the records targeted by this
+    #   harvester.
+    def records
+      batch_size = @threads
+      max_records = @max_records
+      last_record = max_records == 0 ? count : [count, max_records].min
+
+      (0...last_record - 1).step(batch_size).lazy.flat_map do |offset|
+        enumerate_records(records_mets[offset, batch_size])
+      end
+    end
+
+    ##
+    # @param identifier [#to_s] the identifier of the record to get
+    # @return [#to_s] the record
+    def get_record(identifier)
+      enumerate_records(collection_mets
+                          .xpath("//mets:dmdSec[@ID=\"#{identifier}\"]")).first
+    end
+
+    private
+
+    ##
+    # Get a batch of records
+    # @param mets [Nokogiri] the parsed mets for the records to get
+    # @return [Array] an array of @record_class instances
+    def enumerate_records(mets)
+      batch = []
+      mets.each do |rec|
+        uri = rec.xpath('mets:mdRef').first.attribute('href').value
+        batch << { record_uri: uri,
+                   request: @http.add_request(uri: URI.parse(uri)) }
+      end
+
+      batch.flat_map do |record|
+        record[:request].with_response do |response|
+          unless response.status == 200
+            msg = "Couldn't get record #{record[:id]}"
```

Comment:
This actually isn't reporting the record ID that fails - I'm seeing a log message like the following: `E, [2016-03-15T23:10:10.503649 #3766] ERROR -- : [2016-03-15 23:10:10 -0400] [3766] [#<Krikri::Logger:0x007fa34c96f658
>] Couldn't get record`